### PR TITLE
adding condition to run LastModified test

### PR DESCRIPTION
**Pull request by martiansideofthemoon**
I hope the changes are okay @jgraham . Could someone tell me the reason why I have the extra diff under tools? Is it because I didn't run `git submodule update --init --recursive` again after pulling from `upstream` ? Trying to solve #2300

### CODE REVIEW

File: html/dom/documents/resource-metadata-management/document-lastModified-01.html [hunk 1]
```diff
@@ -25,14 +25,22 @@
...
-    var local_time = hours * 60 * 60 + minutes * 60 + seconds;
-    var m = pattern.exec(last_modified);
-    var last_modified_time = parseInt(m[1]) * 60 * 60 + parseInt(m[2]) * 60 + parseInt(m[3]);
-    assert_approx_equals(last_modified_time, local_time, 2,
-                         "Hours and minutes should match local time.");
-  }, "Date returned by lastModified is in the user's local time zone.");
-
+  // Checking whether d and and new instance of Date have the same timezone.
+  // Do not run the time zone test in the case daylight saving occurs.
+  var d2 = new Date();
```
Comment (Manishearth) [inline html/dom/documents/resource-metadata-management/document-lastModified-01.html]: probably should move these two statements after the next two ones. shouldn't affect anything, but is clearer when `date_now` is the last thing executed, and future-proof

File: html/dom/documents/resource-metadata-management/document-lastModified-01.html [hunk 2]
```diff
@@ -25,14 +25,22 @@
...
+  var time_now = d2.getHours() * 60 * 60 + d2.getMinutes() * 60 + d2.getSeconds();
+  var m = pattern.exec(last_modified);
+  var last_modified_time = parseInt(m[1]) * 60 * 60 + parseInt(m[2]) * 60 + parseInt(m[3]);
+
+  if (Math.abs(time_now-last_modified_time)<=2) {
```
Comment (Manishearth) [inline html/dom/documents/resource-metadata-management/document-lastModified-01.html]: Check that `.getTimeZoneOffset()` for the two dates matches

**Comment by martiansideofthemoon** [comment]
@Manishearth adding the changes

File: html/dom/documents/resource-metadata-management/document-lastModified-01.html [hunk 3]
```diff
@@ -25,14 +25,19 @@
...
-    var last_modified_time = parseInt(m[1]) * 60 * 60 + parseInt(m[2]) * 60 + parseInt(m[3]);
-    assert_approx_equals(last_modified_time, local_time, 2,
-                         "Hours and minutes should match local time.");
-  }, "Date returned by lastModified is in the user's local time zone.");
-
+  // Checking whether d and and new instance of Date have the same timezone.
+  // Do not run the time zone test in the case daylight saving occurs.
+  var last_modified_date = new Date(last_modified);
+  var d2 = new Date();
+  if (d2.getTimezoneOffset()==last_modified_date.getTimezoneOffset()) {
```
Comment (Manishearth) [inline html/dom/documents/resource-metadata-management/document-lastModified-01.html]: nit: spaces around operators

Comment (martiansideofthemoon) [inline html/dom/documents/resource-metadata-management/document-lastModified-01.html]: @Manishearth removed the nit. Why has the tools file been edited in my commit?

**Comment by Manishearth** [comment]
You messed up the submodule. `cd` to the submodule folder, `git reset --hard d93ad88336e5933b158129596c196d568ae15f82`, `git add` the submodule folder and `git commit --amend`. In the future, ensure there aren't any submodule changes before using `git commit -a` or `git add *`

(Also, squash the rest of your commits)

**Comment by martiansideofthemoon** [comment]
@Manishearth I hope it is okay.

File: html/dom/documents/resource-metadata-management/document-lastModified-01.html [hunk 4]
```diff
@@ -25,14 +25,19 @@
...
-    var last_modified_time = parseInt(m[1]) * 60 * 60 + parseInt(m[2]) * 60 + parseInt(m[3]);
-    assert_approx_equals(last_modified_time, local_time, 2,
-                         "Hours and minutes should match local time.");
-  }, "Date returned by lastModified is in the user's local time zone.");
-
+  // Checking whether d and and new instance of Date have the same timezone.
+  // Do not run the time zone test in the case daylight saving occurs.
+  var last_modified_date = new Date(last_modified);
+  var d2 = new Date();
+  if (d2.getTimezoneOffset() == last_modified_date.getTimezoneOffset()) {
```
Comment (Manishearth) [inline html/dom/documents/resource-metadata-management/document-lastModified-01.html]: No, just check if d2 and d have different timezones

**Comment by Manishearth** [comment]
r+

**Comment by Manishearth** [comment]
merge? @jgraham

**Pull request merged**